services/diversity/src/util/errorHandler.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Wrapper to catch errors from every endpoint. When an error occurs in a lower layer,
+ * depending on the error it is caught and organized to be thrown again an then log it here,
+ * or it's just handled here.
+ *
+ * @param {Object} logger logger instance.
+ * @param {Function} callback the function to execute.
+ */
+module.exports = logger => callback => (
+  async (req, res, next) => {
+    logger.info(`[${req.id()}] received - ${req.href()}`);
+    try {
+      await callback(req, res, next);
+      logger.info(`[${req.id()}] finished - ${req.href()}`);
+    } catch (e) {
+      const code = typeof e.code === 'number' ? e.code : 500;
+      logger.error(e.stack || e.Error || e.message || e);
+      res.send(code, {
+        code,
+        userMsg: e.message || e.userMsg || 'There was an internal error',
```

Comment:
I believe this order was changed in connectivity service, please put `e.userMsg` before `e.message` and also change it in the other services.

Reply:
Ok. Done